iohk-monitoring/src/Cardano/BM/Data/Trace.lhs

```diff
@@ -21,3 +35,43 @@ A |Trace m a| is a |Tracer m (LogObject a)|.
 
 type Trace m a = Tracer m (LogObject a)
 \end{code}
+
+\subsubsection{Transformer for synopsization}
+\label{code:mkSynopsizedTrace}
+\index{mkSynopsizedTrace}
+Build upon |Synopsizer| tracer transformer to suppress repeated messages,
+based on a given transitive similarity criterion.
+
+A summary is printed every |overflow| entries, and at the end of a run
+of similar messages.
+
+Handy predicates to use:  'loTypeEq' and 'loContentEq'.
+
+See the caveats in the |Synopsizer| tracer transformer description.
+\begin{code}
+
+mkSynopsizedTrace
+  :: forall m a. MonadIO m
+  => (LogObject a -> LogObject a -> Bool)
+  -> Int
```

Comment:
I would like the "max counter" (or "overflow") to be handled in the predicate function. Gives more flexibility and simplifies the code.

Reply:
Done.